python/lsst/pipe/tasks/multiBand.py

```diff
@@ -830,10 +830,12 @@ def getSkySourceFootprints(self, mergedList, skyInfo, growDetectedFootprints=0):
 
             x = int(numpy.random.uniform(xmin, xmax))
             y = int(numpy.random.uniform(ymin, ymax))
-            foot = afwDetect.Footprint(afwGeom.PointI(x, y), skySourceRadius, patchBBox)
+            spans = afwGeom.SpanSet.fromShape(int(skySourceRadius),
+                                              offset=(x, y))
+            foot = afwDetect.Footprint(spans, patchBBox)
             foot.setPeakSchema(self.merged.getPeakSchema())
 
-            if not foot.overlapsMask(mask):
+            if not foot.spans.overlaps(afwGeom.SpanSet.fromMask(mask)):
```

Comment:
With that said, in this particular instance it would probably be better to create an object ``maskSpanSet=afwGeom.SpanSet.fromMask(mask)`` before line 827, to avoid recasting it to the exact same object in each iteration of the loop.